### DISCUSSION TITLE
Local Development Environments

### DISCUSSION THREAD

**Discussion by lostcarpark** [LVerhulst4321/PlanZ#24] [General]
What do you use for your local development environment?

I've been using [dev](https://ddev.com/).

It's a docker based environment, which means you can specify versions of components, such as PHP, MySQL, etc, for each project, Each project has a configuration file that you can specify versions you want, and it will just fetch them. (There are other Docker development environments. I previously tried Lando, but I've found ddev easier to set up).

To set up, you check out the Git repository and run `ddev config`, and it will ask a few questions and set up the environment. Then run `ddev start` each time you want to start up that project.

It creates the database, so you put "db" for all the database params, and you can use `ddev mysql < script` to build the initial database. You can also use `ddev logs` to view PHP logs.

I suspect it does a lot more that I haven't figured out yet, but for setting up containers for projects I find it extremely useful.

**Comment by LVerhulst4321** [comment]
I'll take a look at ddev. Sad to say that I don't really have a local development environment. I have been thinking that I will need one real soon now with this project!

**Reply by lostcarpark** [reply]
It definitely beats installing WAMP/LAMP/MAMP locally!

**Comment by lostcarpark** [comment]
Edited to move from Q&A to General. Apparently Q&A is for StackExchange style questions that expect to have an answer marked as "correct", rather than more opiniony "what do you use?" questions.

**Comment by LVerhulst4321** [comment]
I have been considering getting a separate small PC running some flavor of linux - maybe Ubuntu - to be my con development box.

**Reply by lostcarpark** [reply]
Sorry, in my comment below I said I'd no longer recommend "it". I meant I'd no longer recommend Ubuntu.
There are a few choices worth looking at. Pop OS is the one I'm currently running on my new laptop, and I like it  lot.
Linux Mint is a good choice if you're running slightly older hardware, the interface will be more familiar than some for Windows users, although it does have more of a WIndows XP feel than Windows 10.
Finally, the latest version of Fedora has been getting a lot of good reviews, though I haven't tried it yet.

**Comment by lostcarpark** [comment]
I've been using Linux on my main desktop for over a decade, and I have to say I would no longer recommend it as the easiest Linux for beginners. I'm intending to switch away from it as soon as I find time to do a rebuild.
I'm considering a couple of options, but leaning towards Pop! OS at the moment.
Also, my new laptop just arrived!
![image](https://user-images.githubusercontent.com/17686819/161248746-f254f307-2942-4157-876d-7599c94dd46e.png)

**Reply by bcholmes** [reply]
Pretty!

**Reply by LVerhulst4321** [reply]
Nice!

**Comment by lostcarpark** [comment]
First laptop I've owned that you have to open and install the RAM and SSD before turning on!
![image](https://user-images.githubusercontent.com/17686819/161292576-d522b4e2-4ad5-41d6-8e6b-636b5da2e720.png)

**Reply by LVerhulst4321** [reply]
That's unusual.

**Reply by lostcarpark** [reply]
It's the Framework philosophy. Everything is replaceable and upgradeable. All the ports are slide in modules, so you can decide which port types you want, and which side of the case you prefer them on.
You can, of course, buy it fully preassembled, but they give a discount for completing assembly yourself (which just involves opening the case and slotting in the SSD and RAM).
